File: 1431.py
from typing import List
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        result = []
        for i in range(len(candies)):
            temp = candies.copy()
            temp[i] += extraCandies
            if max(temp) == temp[i]:
                result.append(True)
            else:
                result.append(False)

        return result

candies = [2,3,5,1,3]
extraCandies = 3
sol = Solution()
result = sol.kidsWithCandies(candies, extraCandies)

print(result)

File: 1431.cpp
//
// Created by DuLi on 2023/8/29.
//
#include <iostream>
using std::vector;
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> result;
        int maxCandy = 0;
        for (int candy : candies){
            maxCandy = max(candy + extraCandies, candies);
        }
        for (int candy : candies){
            result.push_back(candy + extraCandies >= maxCandy);
        }
        return result
    }
private:
    int max(int t, vector<int> candies){
        int max = candies[0];
        for (int i = 1; i < candies.size(); ++i) {
            if (candies[i] > max){
                max = candies[i];
            }
        }
        return max;
    }
};

int main(){

    Solution sol;
    sol.kidsWithCandies()
}